File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque 

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res=[]
        que=deque()
        if not root: 
            return []
        que.append((root,0))
        height=0
        while que:
            node,lenn=que.popleft()
            print(len(res),lenn)
            if len(res) <= lenn:
                res.append([])
            print(node.val)
            res[lenn].append(node.val)
            height=lenn+1
            if node.left:
                #leftlen+=1
                que.append((node.left,height))
            if node.right:
                #rightlen+=1
                que.append((node.right,height))
            
        return (res)
        
        

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> list= new ArrayList<>();
        Queue<TreeNode> queue=new LinkedList<>();
        if(root==null){
            return list;
        }
        queue.add(root);
        TreeNode node;
        int len;
        while(!queue.isEmpty()){
            len=queue.size();
            List<Integer> levelList = new ArrayList<>();
            for(int i=0; i<len;i++){
            node=queue.remove();
            levelList.add(node.val);
            if(node.left!=null){
                queue.add(node.left);
            }
            if(node.right!=null){
                queue.add(node.right);
            }
            }
            list.add(levelList);
        }
        return list;
    }
}